File: compiler/waveform/WaveformTest.java
public class WaveformTest {
    public static void main(String[] args) {
        double[] triangleWave = {0.0, 1.0, 0.0, -1.0, 0.0};
        Waveform w = new Waveform(triangleWave);
        System.out.println(w.rms());
    }
}

File: compiler/waveform/waveform_test.rb
require 'waveform'

sine_wave = (0..360).map do |degrees|
  radians = degrees * Math::PI / 180.0
  Math.sin radians
end

waveform = Waveform.new sine_wave

puts waveform.rms
# >> 0.706126729736776
